File: stepik-union/src/main/java/org/stepik/core/serialization/SupportedLanguagesConverter.java
package org.stepik.core.serialization;

import org.stepik.core.SupportedLanguages;
import com.thoughtworks.xstream.converters.SingleValueConverter;

import static org.stepik.core.SupportedLanguages.INVALID;

public class SupportedLanguagesConverter implements SingleValueConverter {
    @Override
    public String toString(Object o) {
        return ((SupportedLanguages) o).getTitle();
    }

    @Override
    public Object fromString(String s) {
        SupportedLanguages language = SupportedLanguages.langOfName(s);
        if (language != INVALID) {
            return language;
        }

        return SupportedLanguages.langOfTitle(s);
    }

    @Override
    public boolean canConvert(Class aClass) {
        return aClass == SupportedLanguages.class;
    }
}

File: stepik-union/src/main/java/org/stepik/core/serialization/SupportedLanguagesConverter.kt
package org.stepik.core.serialization

import org.stepik.core.SupportedLanguages
import com.thoughtworks.xstream.converters.SingleValueConverter

import org.stepik.core.SupportedLanguages.INVALID

class SupportedLanguagesConverter : SingleValueConverter {
    override fun toString(o: Any): String {
        return (o as SupportedLanguages).title
    }

    override fun fromString(s: String): Any {
        val language = SupportedLanguages.langOfName(s)
        if (language != INVALID) {
            return language
        }

        return SupportedLanguages.langOfTitle(s)
    }

    override fun canConvert(aClass: Class<*>): Boolean {
        return aClass == SupportedLanguages::class.java
    }
}